File: ADT/graphs/detect_cycle_1.js
function isCyclicDFS(g, start, visited, parent){
    visited[start] = true
    for(const neighbor of g[start]){
        if(neighbor === parent) continue
        if(visited[neighbor]) return true
        if(isCyclicDFS(g, neighbor, visited, start))
            return true
    }
    return false
}


function isCyclic(g, V){
    const visited = new Array(V).fill(false)
    for(let i = 0; i < V; i++){
        if(!visited[i] && isCyclicDFS(g, i, visited, -1)){
            return true
        }
    }
    return false
}

File: ADT/graphs/detect_cycle_1.py
def is_cyclic_dfs(g, start, visited, parent):
    visited[start] = True
    for neighbor in g[start]:
        if neighbor == parent:
            continue
        if visited[neighbor]:
            return True
        if is_cyclic_dfs(g, neighbor, visited, start):
            return True
    return False
def is_cyclic(g, V):
    visited = [False] * V
    for i in range(V):
        if not visited[i] and is_cyclic_dfs(g, i, visited, -1):
            return True
    return True
        